tidy3d/plugins/mode/mode_solver.py

```diff
@@ -339,6 +353,339 @@ def data_raw(self) -> ModeSolverData:
 
         return mode_solver_data
 
+    @cached_property
+    def bend_axis_3d(self) -> Axis:
+        """Transform the 2D bend axis to its corresponding 3D axis."""
+        _, idx_plane = self.plane.pop_axis((0, 1, 2), axis=self.normal_axis)
+        return idx_plane[self.mode_spec.bend_axis]
+
+    @cached_property
+    def rotated_mode_solver_data(self) -> ModeSolverData:
+        # Create a mode solver with rotated geometries for a 0-degree angle solution
+        solver_ref = self.rotated_structures_copy
+        solver_ref_data = solver_ref.data_raw
+
+        # Transform the mode solution from Cartesian to cylindrical
+        solver_ref_data_cylindrical = self._car_2_cyn(mode_solver_data=solver_ref_data)
+
+        # This solver temp is only used to get the desired modal plane coords for interpolation
+        # and better be avoid one unneccessary mode solver simulation.
+        mode_spec = self.mode_spec.updated_copy(bend_angle_rotation=False)
+        solver_temp = self.updated_copy(mode_spec=mode_spec)
+        solver_temp_data = solver_temp.data_raw
+
+        # compute mode solution by rotating the reference data to the monitor plane
+        rotated_mode_data = self._mode_rotation(
+            solver_ref_data_cylindrical=solver_ref_data_cylindrical,
+            solver_temp_data=solver_temp_data,
+        )
+
+        return rotated_mode_data
+
+    @cached_property
+    def rotated_structures_copy(self):
+        """Creates a copy of the original ModeSolver with rotated structures
+        to the simulation and updates the ModeSpec to disable bend correction
+        and reset angles to normal."""
+
+        rotated_structures = self._rotate_structure()
+        rotated_simulation = self.simulation.updated_copy(structures=rotated_structures)
+        rotated_mode_spec = self.mode_spec.updated_copy(
+            bend_angle_rotation=False, angle_theta=0, angle_phi=0
+        )
+
+        return self.updated_copy(simulation=rotated_simulation, mode_spec=rotated_mode_spec)
+
+    def _rotate_structure(self) -> List[Structure]:
+        "rotate the structures intersecting with modal plane by angle theta if bend_correction"
+        "is enabeled for bend simulations."
```

Comment:
Very minor but better to call it `_rotate_structures` since all structures are rotated. Also, use `"""` comment for function docstrings.